### ISSUE TITLE
when i use nested v-for and switch, it run into error

### ISSUE THREAD

**Issue by toxic-johann** [vuejs/vue#3807]
you can see what happen on codepen
https://codepen.io/toxic-johann/pen/JRJgGJ?editors=1111

i try to render a v-for in a component, and then render the component in another v-for.
when i render this, it's normal.
And in most of the time, it switch normally.
However, in the situation in codepen, it throw an error

> [Vue warn]: It seems there are duplicate keys that is causing an update error. Make sure each v-for item has a unique key. 
> index_record.js:4051 Uncaught (in promise) TypeError: Cannot read property 'tag' of undefined(…)

sth bad happen on the vdom.

i try to add key, but it doesn't work.
so finally, i clear the componet, and then render again.like the behavior in success.

i use vue@2.0-rc8, while on the codepen, it use vue@2.0-rc7,. Whatevet, we can see the same problem.
And i use vuex@1.0-rc2 too.
so, is it a bug? or just a feature?

**Comment by posva** [comment]
Please make sure to read the [Issue Reporting Guidelines](https://github.com/vuejs/vue/blob/dev/CONTRIBUTING.md#issue-reporting-guidelines) before opening new issues. The issue list only accepts bug reports and feature requests. Questions should be posted to the [Gitter chat room](https://gitter.im/vuejs/vue),  [forum](http://forum.vuejs.org/) or [StackOverflow](http://stackoverflow.com/tags/vue.js). Thanks.

---

Your example is hard to follow with one letter variable names everywhere. Also, make sure the problem doesn't appear without Vuex before adding it to the example.
You said you tried using keys but they're not in the example.
I think that if you update your example and publish it on the forums, you'll be able to get some help 😉

**Comment by toxic-johann** [comment]
now i remove the vuex on the codepen, and add keys on v-for.
i use data to set the data.And when i use splice on the data, it occurs.

**Comment by toxic-johann** [comment]
What more.After more testing.I found the actual problem is.In the template, i must have only one root element.Or that situation would happen
